core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -276,6 +277,33 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         }
     }
 
+    private fun believesSameReligion(): Boolean? {
```

Comment:
There's no reason for a null, we don't treat it differently, should be just boolean